Gallery component for landing page

om2137 opened the pull request:
**Description**

![gallery](https://github.com/user-attachments/assets/033c60ed-0fa3-4ddd-bf22-8da7263256fa)
Result with new images from slack in Gallery component.

- Used <img/> instead of next's <Image/> component because with <Image/> the grid was not behaving as desired.
- What should be the behaviour of browse button. 
    - Redirect to a new page/open modal with all the images/videos, we can list them from Cloudianry.
    - Redirect to a Google Drive link.
 let me know, will make the necessary changes

CC: @ashmit-coder @Mayaleeeee let me know if there are any changes to made.

**Related issue**
Resolves #264

TenzDelek commented on components/Gallery/gallery.js:
```diff
@@ -0,0 +1,52 @@
+                    />
+                ))}
+            </div>
+            <Button type='button' onClick={()=>{console.log("click")}} className='w-[200px] mt-4 px-10'>
```
remove the console.log

om2137 replied:
removed it in new commit

TenzDelek commented on public/img/gallery/1000149996.jpg:
change the naming, also as there are many images, having it in webp format would be better instead of jpg

om2137 replied:
converted all images to webp. If there is any particular naming scheme that i should follow let me know.

TenzDelek commented:
one more to add, what is the use case of the button? are we making a dedicated page for the gallery? if yes we have to create a new route for that and if no, i don't think the button is necessary here. it is not doing anything. cc  @om2137 @ashmit-coder @AceTheCreator

om2137 commented:
> one more to add, what is the use case of the button? are we making a dedicated page for the gallery? if yes we have to create a new route for that and if no, i don't think the button is necessary here. it is not doing anything. cc  @om2137 @ashmit-coder @AceTheCreator 

In #264 there are two approaches suggested for the button. I have also listed them in PR description.

ashmit-coder commented:
Hey @Mayaleeeee could you look into this pr!

AceTheCreator reviewed (commented):
Beautiful changes @om2137, but i think it should come after the speakers section. 

I think the browse button is pretty cool. @thulieblack, i think you mentioned a public drive link where all the conference media can be accessible. do you mind sharing it?

om2137 commented:
> Beautiful changes @om2137, but i think it should come after the speakers section.
> 
> I think the browse button is pretty cool. @thulieblack, i think you mentioned a public drive link where all the conference media can be accessible. do you mind sharing it?

@AceTheCreator thankyou, will place the gallery component after speakers section and will add redirection to the drive link once link is available.

thulieblack commented:
/ptal

thulieblack commented:
This is looking good! let's hear what others think

AceTheCreator reviewed (changes requested):
@om2137, let the gallery come after the ticket section

om2137 commented:
> @om2137, let the gallery come after the ticket section

Done, moved gallery after ticket component

ashmit-coder commented:
Hey @AceTheCreator dont we want a href in the navbar as well?

ashmit-coder commented:
@om2137 add a cypress tesf for this section as well .

thulieblack commented:
@om2137 resolve your conflicts and also add this link for the other photos https://drive.google.com/drive/folders/15QooKSy__jerOtLSkXzuKQh0hdlhZqo7?usp=drive_link

om2137 commented:
@thulieblack Done, added the link to button and also resolved the conflict.

om2137 commented:
> @om2137 add a cypress tesf for this section as well .

hey @ashmit-coder I have added the test. please have a look and let me know if there should be any changes to it.

om2137 commented:
@thulieblack I believe all the necessary points for this PR have been covered.
I have added the Cypress test and added a drive link to the browser button. 
are there any other changes?

cc: @AceTheCreator @Mayaleeeee

ashmit-coder commented:
@om2137 the tests are failing do look into it.

ashmit-coder commented:
Please verify your tests before making pings to maintainers.

om2137 commented:
> Please verify your tests before making pings to maintainers.

the test failing is footer.cs.js the changes I made were to landing.cs.js. 
anyways i am looking into it 
will update it soon

ashmit-coder commented on components/Gallery/gallery.js:
```diff
@@ -0,0 +1,56 @@
+                    />
+                ))}
+            </div>
+            <a data-test="gallery-button" href="https://drive.google.com/drive/folders/15QooKSy__jerOtLSkXzuKQh0hdlhZqo7?usp=drive_link" >
+                <Button type='button'className='w-[200px] my-4 px-10'>
+                    Browse
+                </Button>
+            </a>
```
Hey @om2137 maybe open it in a new tab?

ashmit-coder replied:
Other than this the changes LG!

om2137 replied:
> Hey @om2137 maybe open it in a new tab?

done

thulieblack commented:
@AceTheCreator your review

AceTheCreator reviewed (changes requested):
Looks good to me! But I think we should implement lazy loading for the images, and maybe show a skeleton placeholder(The placeholder should imitate the exact width and height of the images to be rendered) while they’re loading — especially for folks on slower networks.

om2137 commented:
> Looks good to me! But I think we should implement lazy loading for the images, and maybe show a skeleton placeholder(The placeholder should imitate the exact width and height of the images to be rendered) while they’re loading — especially for folks on slower networks.


https://github.com/user-attachments/assets/82b50bdc-c4a0-46de-b6dc-dff2976a0e80

@AceTheCreator this skeleton looks good?

om2137 commented:
@AceTheCreator @ashmit-coder I have pushed the necessary changes with lazyloading images with skeleton loading state and browse button opens new tab with drive link

https://github.com/user-attachments/assets/5ba08da9-d598-477b-b75f-e2012237d974

thulieblack commented:
I'm closing this PR no work has been done since
